packages/autocomplete-plugin-recent-searches/src/createPlugin.ts

```diff
@@ -0,0 +1,66 @@
+import { AutocompletePlugin } from '@algolia/autocomplete-core';
+
+import { createRecentSearchStore } from './createRecentSearchStore';
+
+type PluginOptions = {
+  limit?: number;
+};
+
+type RecentSearchesPlugin<TItem> = AutocompletePlugin<TItem> & {
+  getFacetFilters: () => string[];
+};
+
+export function createPlugin<TItem>({
+  limit = 5,
+}: PluginOptions = {}): RecentSearchesPlugin<TItem> {
+  const store = createRecentSearchStore({
+    key: 'RECENT_SEARCHES',
+    limit,
+  });
+
+  return {
+    getSources: ({ query }) => {
+      return query
+        ? []
+        : [
+            {
+              getInputValue: ({ suggestion }) => suggestion.query,
+              getSuggestions() {
+                return store.getAll();
+              },
+              onSelect({ suggestion }) {
+                store.add(suggestion);
+              },
+              templates: {
+                item({ item }) {
```

Comment:
We'll do that in a stacked PR, but we should leverage the `root` parameter to append children to it and attach events on a button to remove the recent search.

Reply:
Oh I didn't know about `root` parameter. Now that I'm seeing the code around it, I'm wondering how users are supposed to attach events on a button in `item()` template?
From where they attach an event and where they can detach it?

Reply:
```ts
const button = document.createElement('button');
button.addEventListener('click', fn);
```

They don't need to detach events because the DOM element is destroyed at every render.

Reply:
yeah but that 

```js
const button = document.createElement('button');
button.addEventListener('click', fn);

root.appendChild(button);
```

should be called after `templates.item()` is evaluated and rendered. Is there a lifecycle hook given to userland?

Reply:
If you don't return any string in `item`, it's not overridden. Try it 🙂

```ts
const search = autocomplete({
  // ...
  templates: {
    item({ item, root }) {
      const button = document.createElement('button');
      button.addEventListener('click', fn);

      root.appendChild(button);
    },
  },
});
```